packages/moonstone/Scrollable/Scrollable.js

```diff
@@ -666,49 +666,99 @@ class ScrollableBase extends Component {
 		}
 	}
 
+	notifyVoiceException = (e, type) => {
+		if (window.webOSVoiceReportActionResult) {
+			window.webOSVoiceReportActionResult({'voiceUi': {'exception': type}});
+			e.preventDefault();
+		}
+	}
+
+	canVoiceScroll = (e) => {
+		const
+			scroll = e && e.detail && e.detail.scroll,
+			{scrollTop, scrollLeft} = this.uiRef,
+			isRtl = this.uiRef.state.rtl;
+		const scrollBounds = this.uiRef.getScrollBounds();
+		const notifyAlreadyCompleted = () => this.notifyVoiceException(e, 'alreadyCompleted');
+
+		if (scroll === 'up' || scroll === 'top') {
+			if (scrollTop === 0) {
+				notifyAlreadyCompleted();
+				return false;
+			}
+		} else if (scroll === 'down' || scroll === 'bottom') {
+			if (scrollTop >= scrollBounds.maxTop - 1) {
+				notifyAlreadyCompleted();
+				return false;
+			}
+		} else if (scroll === 'left' || scroll === 'leftmost') {
+			const limit = isRtl ? scrollBounds.maxLeft : 0;
+			if (scrollLeft === limit) {
```

Comment:
As you did, it is needed to check `scrollLeft >= scrollBounds.maxLeft -1` in here as well as the case when `scroll === 'right' || scroll === 'rightmost'`.